pkg/blockbuilder/blockbuilder.go

```diff
@@ -450,15 +459,39 @@ func (b *BlockBuilder) consumePartition(ctx context.Context, partition int32, st
 	for !sectionEndTime.After(cycleEndTime) {
 		logger := log.With(logger, "section_end", sectionEndTime, "offset", state.Commit.At)
 		state, err = b.consumePartitionSection(ctx, logger, builder, partition, state, sectionEndTime, cycleEndOffset)
-		if err != nil {
+		if err == nil {
+			sectionEndTime = sectionEndTime.Add(b.cfg.ConsumeInterval)
+			continue
+		}
+
+		var recInFutureErr *errFirstRecordInFuture
+		if !errors.As(err, &recInFutureErr) {
```

Comment:
You may have tried this, I think this passage would read easier if it were something like:

```
if err != nil {
   if it's an InFuture error {
      err = nil
      ...
   }
   return PartitionState{}, fmt.Errorf("consume partition %d: %w", partition, err)
} else {
   sectionEndTime = sectionEndTime.Add(b.cfg.ConsumeInterval)
}
```

Reply:
Thanks! Did this with a small change - I kept the nested `if` as `if it's *not* an InFuture error {` in the spirit of keeping the `if` small, and also to avoid `else` since there it no return if its an InFuture error.